reframe/core/schedulers/lsf.py

```diff
@@ -30,45 +32,57 @@ def __init__(self):
             f'schedulers/@{self.registered_name}/job_submit_timeout'
         )
 
+    def _format_option(self, var, option):
+        if var is not None:
+            return self._prefix + ' ' + option.format(var)
+        else:
+            return ''
+
     def emit_preamble(self, job):
-        num_tasks_per_node = job.num_tasks_per_node or 1
-        num_nodes = job.num_tasks // num_tasks_per_node
 
         preamble = [
-            self._format_option(f'-J {job.name}'),
-            self._format_option(f'-o {job.stdout}'),
-            self._format_option(f'-e {job.stderr}'),
-            self._format_option(f'-nnodes {num_nodes}')
+            self._format_option(job.name, '-J "{0}"'),
+            self._format_option(job.stdout, '-o {0}'),
+            self._format_option(job.stderr, '-e {0}'),
+            self._format_option(job.num_tasks, '-n {0}'),
+            self._format_option(
+                min(job.num_tasks * job.num_cpus_per_task,
+                    job.num_tasks_per_node * job.num_cpus_per_task),
+                '-R "span[ptile={0}]"'
+            ),
         ]
 
         # add job time limit in minutes
         if job.time_limit is not None:
             preamble.append(
-                self._format_option(f'-W {int(job.time_limit // 60)}')
+                f'{self._prefix} -W {int(job.time_limit // 60)}'
             )
 
+        for opt in job.sched_access:
+            preamble.append(f'{self._prefix} {opt}')
+
         # emit the rest of the options
         options = job.options + job.cli_options
         for opt in options:
             if opt.startswith('#'):
                 preamble.append(opt)
             else:
-                preamble.append(self._format_option(opt))
+                preamble.append(self._prefix + ' ' + opt)
 
-        # change to working dir with cd
-        preamble.append(f'cd {job.workdir}')
+        if job.sched_exclusive_access:
+            preamble.append(f'{self._prefix} -x')
 
         return preamble
 
     def submit(self, job):
-        cmd = f'bsub {job.script_filename}'
-        completed = _run_strict(cmd, timeout=self._submit_timeout)
+        with open(job.script_filename, 'r') as f:
+            completed = subprocess.run(
+                args='bsub', stdin=f, capture_output=True)
```

Comment:
I've done a PR to your fork, so that you can write this as follows and avoid the `decode()` later on:

```suggestion
        with open(job.script_filename, 'r') as fp:
            completed = _run_strict('bsub', stdin=fp)
```